### PR TITLE
Misc improvements

### PR DESCRIPTION
**Pull request by dbut023**
A collection of general changes I'd made that improved the error reporting, fixed a couple of ruby errors etc.

### CODE REVIEW

File: lib/piggly/command/base.rb [hunk 1]
```diff
@@ -68,18 +68,26 @@ def filter(config, index)
...
+		puts "---"
+                o
+              when :-
+                puts "Rejecting procedures"
+                o = s.reject(&pair.last)
+                puts "Rejected #{s.count - o.count} procedures"
+                puts "Rejected:"
+                ((o-s)|(s-o)).each do |x| puts x.name end
+                puts "---"
+                o
```
Comment (dbut023) [inline lib/piggly/command/base.rb]: Added printing out of the procedures selected and rejected.

File: lib/piggly/command/base.rb [hunk 2]
```diff
@@ -110,11 +118,11 @@ def o_connection_name(config)
...
 
       def o_version(config)
-        lambda { puts "piggly #{VERSION} #{VERSION::RELEASE_DATE}"; exit! }
+        lambda {|x| puts "piggly #{VERSION} #{VERSION::RELEASE_DATE}"; exit! }
       end
 
       def o_dry_run(config)
-        lambda { config.dry_run = true }
+        lambda {|x| config.dry_run = true }
       end
```
Comment (dbut023) [inline lib/piggly/command/base.rb]: Later versions of Ruby I've found get upset if a lambda is missing parameters even if they aren't used.

File: lib/piggly/compiler/trace_compiler.rb [hunk 3]
```diff
@@ -30,6 +31,17 @@ def compile(procedure)
...
+            $stdout.puts <<-EXMSG
+            ****
+            Error compiling procedure #{procedure.name}
+            Source: #{procedure.source_path(@config)}
+            Exception Message:
+            #{e.message}
+            ****
+            EXMSG
+          end
+
```
Comment (dbut023) [inline lib/piggly/compiler/trace_compiler.rb]: Added error message for a failed procedure parse; It just dumps the treetops error which can be a bit terse but it still points you in the right place.

**Comment by kputnam** [comment]
These look like good improvements. Thanks for the contribution!

**Pull request merged**